atlassian_jwt_auth/verifier.py

```diff
@@ -69,12 +71,16 @@ def _decode_jwt(self, a_jwt, key_identifier, jwt_key,
             _msg = ("Claims validity, '%s', exceeds the maximum 1 hour." %
                     (_exp - _iat))
             raise ValueError(_msg)
-
         _jti = claims['jti']
-        if _jti in self._seen_jti:
-            raise ValueError("The jti, '%s', has already been used." % _jti)
+        if self._check_jti_uniqueness:
+            self._check_jti(_jti)
+        return claims
+
+    def _check_jti(self, jti):
+        """Checks that the given jti has not been already been used."""
+        if jti in self._seen_jti:
+            raise ValueError("The jti, '%s', has already been used." % jti)
         else:
-            self._seen_jti[_jti] = None
+            self._seen_jti[jti] = None
```

Comment:
I don't really understand `self._seen_jti` at all. This should most certainly be a `set()` instead of an ordered dictionary.

Reply:
@mark-adams it was originally a set but @erikvanzijst proposed that we change it to an `OrderedDict` so that we would remove the specific entries (#57).